### PR TITLE
MM-28530 - Changes are not automatically propagated to system console when a permission is added/removed from a role

### PR DESCRIPTION
**Pull request by hahmadia**
#### Summary

When a role is patched, no websocket event is sent (so we need referesh on webapp).
This PR sends the websocket event and that means you don't need to refresh on front end to see the role take effect.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-28530

### CODE REVIEW

File: app/role.go [hunk 1]
```diff
@@ -125,6 +125,8 @@ func (a *App) PatchRole(role *model.Role, patch *model.RolePatch) (*model.Role,
 		return nil, err
 	}
 
+	a.sendUpdatedRoleEvent(role)
```
Comment (fmunshi) [inline app/role.go:128]: a.UpdateRole is also _sometimes_ sending a role updated event... can we have that function skip the current role on line 212? basically add a condition here
```
	for _, ir := range impactedRoles {
		a.sendUpdatedRoleEvent(ir)
	}
``` 

changed to 

```
	for _, ir := range impactedRoles {
            if ir.Name != role.Name {
		a.sendUpdatedRoleEvent(ir)
            }
	}
```

Also can we change line 172 to be 
`builtInRolesMinusChannelRoles := append(utils.RemoveStringsFromSlice(model.BuiltInSchemeManagedRoleIDs, builtInChannelRoles...), model.NewSystemRoleIDs...)`? to ignore all the channel moderation related logic if its a new system role

Comment (hahmadia) [reply app/role.go:128]: Done!

**Review by fmunshi — APPROVED** [review]

**Review by sbishel — APPROVED** [review]

**Review by srkgupta — CHANGES_REQUESTED** [review]
Hi @hahmadia 
I am trying to test the PR and the issue is still not fixed and I still have to reload the page for role changes to be reflected. Can you please recheck if the fix is tested and is working fine?

**Comment by hahmadia** [comment]
Got on a call with @srkgupta and it seemed a reset was all that was needed. It's now working fine for him and he will continue his testing.

**Comment by srkgupta** [comment]
Sorry for the confusion @hahmadia. It's working fine now. For some reason, it wasn't working when I last tested.

**Review by srkgupta — APPROVED** [review]
Thanks @hahmadia. Tested the PR and the issue is fixed now and is working fine. Approving the PR.

**Pull request merged**